cmd/logger/logger.go

```diff
@@ -214,10 +219,96 @@ func logIf(level Level, err error, msg string,
 	}
 }
 
-func errorIf(err error, msg string, data ...interface{}) {
-	logIf(Error, err, msg, data...)
+// FatalIf :
+func FatalIf(err error, msg string, data ...interface{}) {
+	logIf(Fatal, err, msg, data...)
 }
 
-func fatalIf(err error, msg string, data ...interface{}) {
-	logIf(Fatal, err, msg, data...)
+// LogIf :
+func LogIf(ctx context.Context, err error) {
+	if err == nil {
+		return
+	}
+
+	req := GetReqInfo(ctx)
+	if req == nil {
+		req = &ReqInfo{}
+	}
+	tags := make(map[string]string)
+	for _, entry := range req.GetTags() {
+		tags[entry.Key] = entry.Val
+	}
+
+	// Get the cause for the Error
+	message := err.Error()
+	// Get full stack trace
+	trace := getTrace(2)
+	// Output the formatted log message at console
+	var output string
+	if req.API == "" {
+		req.API = "SYSTEM"
+	}
+	if jsonFlag {
+		logJSON, err := json.Marshal(&logEntry{
+			Level:      Error.String(),
+			RemoteHost: req.RemoteHost,
+			RequestID:  req.RequestID,
+			UserAgent:  req.UserAgent,
+			Time:       time.Now().UTC().Format(time.RFC3339Nano),
+			API:        api{Name: req.API, Args: args{Bucket: req.BucketName, Object: req.ObjectName}},
+			Trace:      traceEntry{Message: message, Source: trace, Variables: tags},
+		})
+		if err != nil {
+			panic("json marshal of logEntry failed: " + err.Error())
```

Comment:
why not simply `panic(err)` there is no reason for custom error string with `panic()`

Reply:
done